From 308181f2bd6c512258c30b9487859767b2b40c0e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 11 Jan 2021 10:22:06 +0000
Subject: [PATCH] rhbz#1913828 SfxViewFrame::Current() can return null

revert cui part of

commit 849b837d1a3b185a8dd893a8f6eaed53605bcab1
Date:   Tue Jun 4 15:49:12 2019 -0800

    tdf#83618 Make line spacing values agree

the reported problem seems still fixed after doing this

Change-Id: I4a3f938b4137e2c36ed450936e06ad1f02979976
---
 cui/source/tabpages/paragrph.cxx | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/cui/source/tabpages/paragrph.cxx b/cui/source/tabpages/paragrph.cxx
index 6ad843bdc9b3..eaa0daf23cfe 100644
--- a/cui/source/tabpages/paragrph.cxx
+++ b/cui/source/tabpages/paragrph.cxx
@@ -51,8 +51,6 @@
 #include <svl/eitem.hxx>
 #include <svl/intitem.hxx>
 
-#include <sfx2/viewfrm.hxx>
-
 const sal_uInt16 SvxStdParagraphTabPage::pStdRanges[] =
 {
     SID_ATTR_PARA_LINESPACE,        // 10033
@@ -232,7 +230,7 @@ bool SvxStdParagraphTabPage::FillItemSet( SfxItemSet* rOutSet )
            m_xLineDistAtMetricBox->get_value_changed_from_saved() ) )
     {
         nWhich = GetWhich( SID_ATTR_PARA_LINESPACE );
-        MapUnit eUnit = SfxViewFrame::Current()->GetPool().GetMetric( nWhich );
+        MapUnit eUnit = pPool->GetMetric( nWhich );
         SvxLineSpacingItem aSpacing(
             static_cast<const SvxLineSpacingItem&>(GetItemSet().Get( nWhich )) );
 
@@ -722,7 +720,7 @@ void SvxStdParagraphTabPage::SetLineSpacing_Impl
     const SvxLineSpacingItem &rAttr
 )
 {
-    MapUnit eUnit = SfxViewFrame::Current()->GetPool().GetMetric( rAttr.Which() );
+    MapUnit eUnit = GetItemSet().GetPool()->GetMetric( rAttr.Which() );
 
     switch( rAttr.GetLineSpaceRule() )
     {
-- 
2.28.0

